crucible-go/src/Lang/Crucible/Go/Simulate.hs

```diff
@@ -0,0 +1,179 @@
+import           Control.Monad.Fail (MonadFail)
+
+import           Data.Text (Text)
+
+import           System.IO
+
+-- parameterized-utils
+import           Data.Parameterized.Context as Ctx
+
+-- crucible
+import qualified Lang.Crucible.Analysis.Postdom as C
+import           Lang.Crucible.Backend
+import           Lang.Crucible.CFG.Core
+import           Lang.Crucible.CFG.Expr as C
+import qualified Lang.Crucible.CFG.Reg as Reg
+import           Lang.Crucible.Simulator.Evaluation
+import           Lang.Crucible.Simulator.ExecutionTree
+import           Lang.Crucible.Simulator.GlobalState
+
+import           Lang.Crucible.FunctionHandle as C
+import qualified Lang.Crucible.Simulator as C
+
+-- crux
+import qualified Crux.Types   as Crux
+
+-- what4
+import qualified What4.Config as W4
+import           What4.FunctionName (FunctionName)
+import qualified What4.Interface as W4
+
+-- go
+import            Language.Go.AST
+import            Language.Go.Desugar (desugar)
+import            Language.Go.Parser as P
+import            Language.Go.Rename (rename)
+import            Language.Go.Types
+
+import            Lang.Crucible.Go.Overrides
+import            Lang.Crucible.Go.Translation (translate)
+import            Lang.Crucible.Go.Types
+
+-- | Borrowed from crucible-jvm.
+setSimulatorVerbosity :: (W4.IsSymExprBuilder sym) => Int -> sym -> IO ()
+setSimulatorVerbosity verbosity sym = do
+  verbSetting <- W4.getOptionSetting W4.verbosity (W4.getConfiguration sym)
+  _ <- W4.setOpt verbSetting (toInteger verbosity)
+  return ()
+
+-- | No syntax extensions.
+goExtensionImpl :: C.ExtensionImpl p sym Go
+goExtensionImpl =
+  C.ExtensionImpl (\_sym _iTypes _logFn _f x -> case x of) (\x -> case x of)
+
+failIfNotEqual :: forall k f m a (b :: k).
+                  (MonadFail m, Show (f a), Show (f b), TestEquality f)
+               => f a -> f b -> String -> m (a :~: b)
+failIfNotEqual r1 r2 str
+  | Just Refl <- testEquality r1 r2 = return Refl
+  | otherwise = fail $ str ++ ": mismatch between "
+                ++ show r1 ++ " and " ++ show r2
+
+mkFunctionBindings :: forall sym p ext.
+                      [SomeOverride p sym ext]
+                   -> [(Maybe (Text, FunctionName), AnyCFG ext)]
+                   -> FunctionBindings p sym ext
+mkFunctionBindings _overrides [] = emptyHandleMap
+mkFunctionBindings overrides ((ident, AnyCFG cfg) : cfgs) =
+  let f = case lookupOverride' ident overrides of
+        Just (SomeOverride _pkg argsRepr retRepr override) ->
+          case (testEquality (cfgArgTypes cfg) argsRepr,
+                testEquality (cfgReturnType cfg) retRepr) of
+            (Just Refl, Just Refl) -> UseOverride override
+            _ -> error $ "mkFunctionBindings: override " ++ show ident ++
+                 " type mismatch. expected " ++ show (cfgArgTypes cfg) ++
+                 " -> " ++ show (cfgReturnType cfg) ++ ", got " ++
+                 show argsRepr ++ " -> " ++ show retRepr
```

Comment:
Maybe now isn't the time, but we have been trying to move to `panic` instead of `error`

Reply:
Done.